flip/uploader/session.go

```diff
@@ -15,42 +17,55 @@ type Session struct {
 
 	// Parts specifies the number of parts the file is supposed to be split into
 	// for uploading purposes.
-	Parts int `json:"parts"`
+	Parts int `json:"parts,string"`
 
 	// PartSize specifies the size of each part. The last part can be smaller
 	// than this value.
-	PartSize int64 `json:"part_size"`
+	PartSize int64 `json:"part_size,string"`
 
 	// MaxConnections specifies the maximum number of possible concurrent
 	// uploads.
-	MaxConnections int `json:"max_connections"`
+	MaxConnections int `json:"max_connections,string"`
+
+	ExtraFiles map[string]extra_files `json:"extra_files"`
 }
 
-// UnmarshalJSON implements the json.Unmarshaler interface. Telestream Cloud's
-// API returns every value as string.
-func (s *Session) UnmarshalJSON(b []byte) error {
-	var m map[string]string
-	if err := json.Unmarshal(b, &m); err != nil {
-		return err
-	}
-	partSize, err := strconv.ParseInt(m["part_size"], 10, 64)
-	if err != nil {
-		return err
-	}
-	parts, err := strconv.Atoi(m["parts"])
-	if err != nil {
-		return err
-	}
-	maxConnections, err := strconv.Atoi(m["max_connections"])
-	if err != nil {
-		return err
+type extra_files struct {
+	// Parts specifies the number of parts the file is supposed to be split into
+	// for uploading purposes.
+	Parts int `json:"parts,string"`
+
+	// PartSize specifies the size of each part. The last part can be smaller
+	// than this value.
+	PartSize int64 `json:"part_size,string"`
+}
+
+type ExtraFileInfo []struct {
+	Tag   string
+	Files []string
+}
+
+func (self *ExtraFileInfo) MarshalJSON() ([]byte, error) {
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#receiver-names

Also, it shouldn't be a pointer receiver